spring-vault-core/src/main/java/org/springframework/vault/support/Plaintext.java

```diff
@@ -138,6 +138,19 @@ public String asString(Charset charset) {
 		return new String(getPlaintext(), charset);
 	}
 
+	/**
+	 * @return <tt>true</tt> if the plaintext is base64 encoded
+	 */
+	public boolean isBase64() {
```

Comment:
I'm not sure this is something we should check. by decoding. The payload can be rather big and decoding it on our side imposes quite some load. If the payload isn't Base64-encoded, then Vault will report the issue in the HTTP response.

Reply:
@mp911de correct, I did not think about the size of the possible plaintext and indeed Vault will report an error. I've updated the PR